hvac/adapters.py

```diff
@@ -91,14 +91,16 @@ def __init__(
             session = requests.Session()
             session.cert, session.verify, session.proxies = cert, verify, proxies
         # fix for issue 991 using session verify if set
-        else:
+        elif self.check_session_object(session):
```

Comment:
just stumbled upon that PR, I really want hvac to be future proof regarding that validation as we're using this in production.

```suggestion
        elif hasattr(session, "adapters") and isinstance(session.adapters, dict):
```

this is largely sufficient.

Reply:
Ok just making sure I am understanding this correctly. All we need to do to verify a session is a valid session object is make sure it has an adapters attribute and that adapters attribute is a dictionary?

Just double checking that this is a universal truth for the other requests compatible libraries. It completely makes sense that it would be that simple. I'm just dotting I's here but I think this sounds true. (It's late but if I don't do it now I will forget I saw it.)

Reply:
> is a valid session object

yes.

> that this is a universal truth for the other requests compatible libraries

to my best knowledge anything that want to be "requests" compatible must implement the concept of "adapters". otherwise it's a breaking change.